PolicyOptimization/DistribPolicyGradients/Client.py

```diff
@@ -117,20 +111,42 @@ def configure(self):
         self.cfg = self.client.get_cfg()
 
         self.env, self.experience, gradient_builder, policy_gradient_optimizer, value_gradient_optimizer, \
-        self.agent, self.policy, self.strategy_optimizer, adaptive_omega, self.value_net, \
+        self.agent, self.policy, self.strategy_optimizer, adaptive_omega, value_net, \
         novelty_gradient_optimizer, learner = Configurator.build_vars(self.cfg, existing_env=env)
 
         self.env.reset()
         self.transmit_env_spaces()
-        self.exp_manager = DistribExperienceManager(self.cfg, client=self.client)
         self.last_checked = time.time()
 
         print("Fetching initial models...")
         while not self.update_models():
             time.sleep(1)
 
+        tfh_cfg = self.cfg.copy()
+        tfh_cfg["env_space_shapes"] = (self.policy.input_shape, self.policy.output_shape)
+        self.trajectory_finalizer_handler = MPFProcessHandler()
+        self.trajectory_finalizer_handler.setup_process(ClientTrajectoryFinalizer())
+        self.trajectory_finalizer_handler.put(ClientTrajectoryFinalizer.HEADER_INITIALIZATION, tfh_cfg)
+
         print("Client configuration complete!")
 
     def cleanup(self):
-        if self.client is not None:
-            self.client.disconnect()
+        try:
+            if self.client is not None:
+                self.client.disconnect()
+        except:
+            pass
+
+        try:
+            if self.env is not None:
+                self.env.close()
+        except:
+            pass
+
+        try:
+            if self.trajectory_finalizer_handler is not None:
+                self.trajectory_finalizer_handler.put(ClientTrajectoryFinalizer.HEADER_FLUSH,
+                                                      data=self.agent.ep_rewards if self.agent and len(self.agent.ep_rewards) > 0 else None)
+                self.trajectory_finalizer_handler.close()
+        except:
+            pass
```

Comment:
TODO before merge: add a `.editorconfig` so this stops happening

Reply:
I actually think a `.editorconfig` belongs in its own PR, so I'm resolving this for now.